test/src/test/java/hudson/model/QueueTest.java

```diff
@@ -780,4 +782,24 @@ public void run() {
                 "B finished at " + buildBEndTime + ", C started at " + buildC.getStartTimeInMillis(), 
                 buildC.getStartTimeInMillis() >= buildBEndTime);
     }
+
+    @Issue("JENKINS-30084")
+    @Test
+    /*
+     * When a flyweight task is restricted to run on a specific node, the node will be provisioned
+     * and the flyweight task will be executed.
+     */
+    public void flyWeightTaskQueue () throws Exception {
```

Comment:
please name test to show:
- what the main action of this test
- what we should expect

usually the word `should` at the start and word `when` in the middle helps to do so.

For example = `shouldRunFlyweightTaskOnProvisionedNodeWhenNodeRestricted`

it can save a lot of time if this test will fail during CI when you don't know what the hell is that